Detectors/TPC/simulation/include/TPCSimulation/GBTFrameContainer.h

```diff
@@ -0,0 +1,233 @@
+    /// @param s2hw0 half-word 0 from SAMPA 2
+    /// @param s2hw1 half-word 1 from SAMPA 2
+    /// @param s2hw2 half-word 2 from SAMPA 2
+    /// @param s2hw3 half-word 3 from SAMPA 2
+    /// @param s0adc ADC clock from SAMPA 0
+    /// @param s1adc ADC clock from SAMPA 1
+    /// @param s2adc ADC clock from SAMPA 2
+    /// @param marker additional 16 bit marker which is not part of the actual frame
+    void addGBTFrame(short& s0hw0l, short& s0hw1l, short& s0hw2l, short& s0hw3l,
+                     short& s0hw0h, short& s0hw1h, short& s0hw2h, short& s0hw3h,
+                     short& s1hw0l, short& s1hw1l, short& s1hw2l, short& s1hw3l,
+                     short& s1hw0h, short& s1hw1h, short& s1hw2h, short& s1hw3h,
+                     short& s2hw0,  short& s2hw1,  short& s2hw2,  short& s2hw3, 
+                     short& s0adc,  short& s1adc,  short& s2adc,  unsigned marker = 0);
+
+    template<typename... Args> void addGBTFrame(Args&&... args);
+
+    /// Add all frames from file to conatiner
+    /// @param fileName Path to file
+    void addGBTFramesFromFile(std::string fileName);
+
+    /// Add all frames from file to conatiner
+    /// @param fileName Path to file
+    void addGBTFramesFromBinaryFile(std::string fileName);
+
+//    /// Fill output TClonesArray
+//    /// @param output Output container
+//    void fillOutputContainer(TClonesArray* output);
+
+    /// Set enable the ADC clock warnings
+    /// @param val Set it to true or false
+    void setEnableAdcClockWarning(bool val)     { mEnableAdcClockWarning = val; };
+
+    /// Set enable the sync pattern position warnings
+    /// @param val Set it to true or false
+    void setEnableSyncPatternWarning(bool val)  { mEnableSyncPatternWarning = val; };
+
+    /// Set enable compilation of ADC values
+    /// @param val Set it to true or false
+    void setEnableCompileAdcValues(bool val)    { mEnableCompileAdcValues = val; };
+
+    /// Option to store the inserted GBT frames
+    /// Option to store the inserted GBT frames
+    /// @param val Set it to true or false
+    void setEnableStoreGBTFrames(bool val)      { mEnableStoreGBTFrames = val; };
+
+    /// Extracts the digits after all 32 channel were transmitted
+    /// @param digitContainer Digit Container to store the digits in
+    /// @return If true, at least one digit was added.
+    bool getDigits(std::vector<Digit> *digitContainer, bool removeChannel = true);
+
+    bool getData(std::vector<HalfSAMPAData> *container, bool removeChannel = true);
+
+    int getNFramesAnalyzed() const { return mGBTFramesAnalyzed; };
+
+    /// Returns a GBT frame
+    /// @param index
+    /// @return GBT frame
+    GBTFrame getGBTFrame(int index) const { return mGBTFrames[index]; };
+    GBTFrame operator[] (int index) const { return mGBTFrames[index]; };
+    GBTFrame& operator[] (int index)      { return mGBTFrames[index]; };
+
+    std::vector<GBTFrame>::iterator begin()               { return mGBTFrames.begin(); };
```

Comment:
I am wondering if it would be more convenient to return a reference to `mGBTFrames` and then the caller also has the iterators directly.